File: android/src/main/java/top/huic/tencent_im_plugin/listener/CustomSignalingListener.java
package top.huic.tencent_im_plugin.listener;

import com.tencent.imsdk.v2.V2TIMSignalingListener;

import java.util.HashMap;
import java.util.List;

import top.huic.tencent_im_plugin.TencentImPlugin;
import top.huic.tencent_im_plugin.enums.ListenerTypeEnum;

/**
 * 自定义信令监听器
 */
public class CustomSignalingListener extends V2TIMSignalingListener {
    /**
     * 收到新邀请时
     */
    @Override
    public void onReceiveNewInvitation(final String inviteID, final String inviter, final String groupID, final List<String> inviteeList, final String data) {
        super.onReceiveNewInvitation(inviteID, inviter, groupID, inviteeList, data);
        TencentImPlugin.invokeListener(ListenerTypeEnum.ReceiveNewInvitation, new HashMap<String, Object>() {
            {
                put("inviteID", inviteID);
                put("inviter", inviter);
                put("groupID", groupID);
                put("inviteeList", inviteeList);
                put("data", data);
            }
        });
    }

    /**
     * 被邀请者接受邀请
     */
    @Override
    public void onInviteeAccepted(final String inviteID, final String invitee, final String data) {
        super.onInviteeAccepted(inviteID, invitee, data);
        TencentImPlugin.invokeListener(ListenerTypeEnum.InviteeAccepted, new HashMap<String, Object>() {
            {
                put("inviteID", inviteID);
                put("invitee", invitee);
                put("data", data);
            }
        });
    }

    /**
     * 被邀请者拒绝邀请
     */
    @Override
    public void onInviteeRejected(final String inviteID, final String invitee, final String data) {
        super.onInviteeRejected(inviteID, invitee, data);
        TencentImPlugin.invokeListener(ListenerTypeEnum.InviteeRejected, new HashMap<String, Object>() {
            {
                put("inviteID", inviteID);
                put("invitee", invitee);
                put("data", data);
            }
        });
    }

    /**
     * 邀请被取消
     */
    @Override
    public void onInvitationCancelled(final String inviteID, final String inviter, final String data) {
        super.onInvitationCancelled(inviteID, inviter, data);
        TencentImPlugin.invokeListener(ListenerTypeEnum.InvitationCancelled, new HashMap<String, Object>() {
            {
                put("inviteID", inviteID);
                put("inviter", inviter);
                put("data", data);
            }
        });
    }

    /**
     * 邀请超时
     */
    @Override
    public void onInvitationTimeout(final String inviteID, final List<String> inviteeList) {
        super.onInvitationTimeout(inviteID, inviteeList);
        TencentImPlugin.invokeListener(ListenerTypeEnum.InvitationTimeout, new HashMap<String, Object>() {
            {
                put("inviteID", inviteID);
                put("inviteeList", inviteeList);
            }
        });
    }
}

File: ios/Classes/listener/CustomSignalingListener.swift
//
// Created by 蒋具宏 on 2020/10/27.
//

import Foundation
import ImSDK_Plus

/// 自定义信令监听
class CustomSignalingListener: NSObject, V2TIMSignalingListener {
    ///收到新邀请时
    func onReceiveNewInvitation(_ inviteID: String!, inviter: String!, groupID: String!, inviteeList: [String]!, data: String?) {
        var params: [String: Any] = ["inviteID": inviteID!];
        if let temp = inviter {
            params["inviter"] = temp;
        }
        if let temp = groupID {
            params["groupID"] = temp;
        }
        if let temp = inviteeList {
            params["inviteeList"] = temp;
        }
        if let temp = data {
            params["data"] = temp;
        }
        SwiftTencentImPlugin.invokeListener(type: ListenerType.ReceiveNewInvitation, params: params)
    }

    /// 被邀请者接受邀请
    func onInviteeAccepted(_ inviteID: String!, invitee: String!, data: String?) {
        var params: [String: Any] = ["inviteID": inviteID!,"invitee": invitee!];
        if let temp = data {
            params["data"] = temp;
        }
        SwiftTencentImPlugin.invokeListener(type: ListenerType.InviteeAccepted, params: params)
    }

    /// 被邀请者拒绝邀请
    func onInviteeRejected(_ inviteID: String!, invitee: String!, data: String?) {
        var params: [String: Any] = ["inviteID": inviteID!,"invitee": invitee!];
        if let temp = data {
            params["data"] = temp;
        }
        SwiftTencentImPlugin.invokeListener(type: ListenerType.InviteeRejected, params: params)
    }

    /// 邀请被取消
    func onInvitationCancelled(_ inviteID: String!, inviter: String!, data: String?) {
        var params: [String: Any] = ["inviteID": inviteID!,"inviter": inviter!];
        if let temp = data {
            params["data"] = temp;
        }
        SwiftTencentImPlugin.invokeListener(type: ListenerType.InvitationCancelled, params: params)
    }

    /// 邀请超时
    func onInvitationTimeout(_ inviteID: String!, inviteeList: [String]!) {
        SwiftTencentImPlugin.invokeListener(type: ListenerType.InvitationTimeout, params: [
            "inviteID": inviteID!,
            "inviteeList": inviteeList!,
        ])
    }
}